src/Calculator/SettingsPage.xaml.cpp

```diff
@@ -0,0 +1,208 @@
+
+    ContributeRunBeforeLink->Text = contributeTextBeforeHyperlink;
+    ContributeRunLink->Text = contributeTextLink;
+    ContributeRunAfterLink->Text = contributeTextAfterHyperlink;
+
+}
+
+void SettingsPage::BackButtonClick(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto rootFrame = dynamic_cast<::Frame ^>(Window::Current->Content);
+
+    if (SettingsRestartApp->Visibility == ::Visibility::Visible)
+    {
+        Windows::Storage::ApplicationData::Current->LocalSettings->Values->Insert(L"restartApp", L"True");
+    }
+    else
+    {
+        Windows::Storage::ApplicationData::Current->LocalSettings->Values->Insert(L"restartApp", L"False");
+    }
+    rootFrame->Navigate((MainPage::typeid), this);
+}
+
+void SettingsPage::SetVersionString()
+{
+    PackageVersion version = Package::Current->Id->Version;
+    String ^ appName = AppResourceProvider::GetInstance()->GetResourceString("AppName");
+
+    SettingsVersion->Text = appName + L" " + version.Major + L"." + version.Minor + L"." + version.Build + L"." + version.Revision;
+}
+
+void SettingsPage::SetCopyrightString()
+{
+    auto resourceLoader = AppResourceProvider::GetInstance();
+
+    auto copyrightText = LocalizationStringUtil::GetLocalizedString(resourceLoader->GetResourceString("SettingsCopyright"), StringReference(to_wstring(BUILD_YEAR).c_str()));
+
+    SettingsCopyright->Text = copyrightText;
+}
+
+void SettingsPage::ColorSettingsButtonClicked(_In_ Object ^ sender, _In_ RoutedEventArgs ^ e)
+{
+    Launcher::LaunchUriAsync(ref new Uri("ms-settings:colors"));
+}
+
+void SettingsPage::SettingsFeedbackButtonClick(_In_ Object ^ sender, _In_ RoutedEventArgs ^ e)
+{
+    PackageVersion version = Package::Current->Id->Version;
+    String ^ versionNumber = ref new String(L"Version ");
+    versionNumber = versionNumber + version.Major + version.Minor + version.Build + version.Revision;
+
+    Launcher::LaunchUriAsync(ref new Uri("windows-feedback:?contextid=130&metadata=%7B%22Metadata%22:[%7B%22AppBuild%22:%22" + versionNumber + "%22%7D]%7D"));
+}
+
+void SettingsPage::ThemeChecked(Platform::Object ^ sender, Windows::UI::Xaml::RoutedEventArgs ^ e)
+{
+    auto resourceLoader = AppResourceProvider::GetInstance();
+    auto currentThemeValue = Windows::Storage::ApplicationData::Current->LocalSettings->Values->Lookup(L"CurrentTheme");
+    RadioButton ^ radioButton = safe_cast<RadioButton ^>(sender);
+    String ^ tag = radioButton->Tag->ToString();
+    String ^ currentTheme = safe_cast<String ^>(currentThemeValue);
+
+    // Only display the restart app notice if the theme selected is not the current theme.
+    if (currentTheme == tag)
+    {
+        SettingsRestartApp->Visibility = ::Visibility::Collapsed;
+    }
+    else
+    {
+        SettingsRestartApp->Visibility = ::Visibility::Visible;
+        SettingsRestartApp->Text = resourceLoader->GetResourceString("SettingsRestartNotice");
```

Comment:
Can you set the Text in XAML and then just control visibility in code behind instead so that it only gets set once?